src/ergw_dns.erl

```diff
@@ -0,0 +1,95 @@
+%% Copyright 2018, Travelping GmbH <info@travelping.com>
+
+%% This program is free software; you can redistribute it and/or
+%% modify it under the terms of the GNU General Public License
+%% as published by the Free Software Foundation; either version
+%% 2 of the License, or (at your option) any later version.
+
+%% @doc
+%% This modules implements DNS-based Node Discovery according to 3GPP TS 29.303.
+%% Right now this makes decission by lookup NAPTR record for given name. This 
+%% understands service and protocol names which are described in section 
+%% 19.4.3 of 3GPP TS 23.003. 
+
+-module(ergw_dns).
+
+-export([naptr/1, lookup/1, lookup/2]).
+
+%%====================================================================
+%% API
+%%====================================================================
+
+%% @private
+naptr(Name) ->
```

Comment:
Sorry, for not noticing earlier.

Erlang DNS will by default use UDP and not retry with TCP if the answer packet gets truncated.
So please, change that to
```
naptr(Name) ->
    NsOpts =
	case setup:get_env(ergw, nameservers) of
	    undefined ->
		[];
	{ok, Nameservers} ->
		[{nameservers, Nameservers}]
	end,
    %% 3GPP DNS answers are almost always large, use TCP by default....
    inet_res:resolve(Name, in, naptr, [{usevc, true} | NsOpts]).
```

Reply:
Changed